codecarbon/external/scheduler.py

```diff
@@ -33,6 +34,7 @@ def start(self, from_run=False):
         if from_run or self._stopped:
             self._stopped = False
             self._timer = Timer(self.interval, self._run)
+            self._timer.daemon = True
```

Comment:
@benoit-cty this parameter makes the scheduler stop when an exception occurs in the main thread. 
The tracker now is able to shutdown gracefully when used as decorator or context manager. I think we should discourage the direct usage of start & stop functions, in the main or task-based tracker.